utils/npm/version.go

```diff
@@ -0,0 +1,66 @@
+package npmutils
+
+import (
+	"errors"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+	"github.com/jfrog/jfrog-client-go/utils/version"
+	"io"
+	"io/ioutil"
+	"os/exec"
+
+	gofrogcmd "github.com/jfrog/gofrog/io"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+)
+
+func Version(executablePath string) (*version.Version, error) {
+	pipeReader, pipeWriter := io.Pipe()
+	defer pipeReader.Close()
+	defer pipeWriter.Close()
```

Comment:
The error returned from the above two Close() methods is ignored.
Here's how we can avoid ignoring this error, which inside a defer clause:
1. Change the function declaration to using named returned values - 
```
func Version(executablePath string) (version *version.Version, err error) {
```
2. Change the defer, so thay the error returned from ```pipeReader.Close()``` is returned, if the function did not return an error - 
```
	defer func() {
		e := pipeReader.Close()
		if err == nil {
			err = e
		}
		e = pipeWriter.Close()
		if err == nil {
			err = e
		}
	}()
```
Let's go over the code base and make this change in other places as well.